### ISSUE TITLE
[bug] Logged users can view each other encrypted files in plaintext

### ISSUE THREAD

**Issue by ghost** [google/fscrypt#61]
I noticed that logged user can view files from other logged users encrypted directories even as they see encrypted directory status as locked and don't have relevant key in their keyring. Is it side effect of linking user keyring to root's? I tested this on Archlinux with `fscrypt --version 0.2.1-1-ga949b13` while logging in with VT console. Also all users in Arch share the same GID "users" by default. After user logout directory is properly visible as encrypted.

**Encrypted directory status viewed as user2:**
```
$ fscrypt status /home/user1
"/home/user1" is encrypted with fscrypt.

Policy:   949471831dcf55cf
Unlocked: No

Protected with 1 protector:
PROTECTOR         LINKED  DESCRIPTION
6682ae84e70e99b3  No      login protector for user1
```
**User2 keyring:**
```
$ keyctl show
Session Keyring
 168271410 --alswrv   1002  1002  keyring: _ses
 840784887 --alswrv   1002 65534   \_ keyring: _uid.1002
 800661253 --alsw-v   1002  1002       \_ logon: ext4:13cb92d62226353b
```
**Encrypted directory view as user2:**
```
$ ls -al /home/user1/
total 12
drwxr-xr-x 2 user1 user1 4096 Sep  3 16:14 .
drwxr-xr-x 6 root  root  4096 Sep  3 16:10 ..
-rw-r--r-- 1 user1 user1  225 Sep  8 12:02 .bash_history
```
**Encrypted file view as user2:**
```
$ cat /home/user1/.bash_history
ls -al
fscrypt encrypt /home/user1
exit
exit
fscrypt status /home/user1/
exit
exit
exit
exit
chmod 644 .bash_history 
logout
exit
exit
exit
```

**Comment by Redsandro** [comment]
Unfortunately this appears to be a side-effect from the current implementation.

@ebiggers [said](https://github.com/google/fscrypt/issues/111#issuecomment-444352798): 

> This is a known issue with filesystem encryption in the kernel: the caches that allow encrypted files to be accessed are systemwide

So your data is safe from extraction after physical hardware theft. However, parties that manage to get some kind of access to your machine can still spy on your files. As soon as you access them, they can see them too.

**Comment by ebiggers** [comment]
As mentioned on other issues, you can use `chmod 0700` to forbid other users from accessing directories.

**Comment by Redsandro** [comment]
Just as a reminder for people who come here from google: __Make sure to do a `umask 077` that sticks.__

Otherwise every new dir/file created with your user will default to 755/644.

Be aware of the maze that is `.profile`, `.bash_profile`, `.bashrc` and which one is read when. It might be best to set system-wide defaults ~~in `/etc/profile`~~ in `/etc/login.defs`.

**Comment by ebiggers** [comment]
https://github.com/google/fscrypt/commit/2b6667a677bf80be0f7dd14baa2516bfdd1f0b50 changed `fscrypt encrypt` to set mode `0700` on new encrypted directories.

So I think this issue should be closed.

**Comment by josephlr** [comment]
Closing as resolved by #134
